dbms/src/Flash/Coprocessor/DAGQuerySource.cpp

```diff
@@ -39,21 +34,8 @@ DAGQuerySource::DAGQuerySource(Context & context_)
         root_query_block = std::make_shared<DAGQueryBlock>(1, dag_request.executors());
     }
 
-    root_query_block->output_field_types = collectOutputFieldTypes(dag_request);
     getDAGContext().initExecutorIdToJoinIdMap();
-
-    for (UInt32 i : dag_request.output_offsets())
-    {
-        root_query_block->output_offsets.push_back(i);
-        if (unlikely(i >= root_query_block->output_field_types.size()))
-            throw TiFlashException(
-                fmt::format("{}: Invalid output offset(schema has {} columns, access index {}", __PRETTY_FUNCTION__, root_query_block->output_field_types.size(), i),
-                Errors::Coprocessor::BadRequest);
-        getDAGContext().result_field_types.push_back(root_query_block->output_field_types[i]);
-    }
-    auto encode_type = analyzeDAGEncodeType(getDAGContext());
-    getDAGContext().encode_type = encode_type;
-    getDAGContext().keep_session_timezone_info = encode_type == tipb::EncodeType::TypeChunk || encode_type == tipb::EncodeType::TypeCHBlock;
+    getDAGContext().initOutputInfo();
```

Comment:
Since all the information is moved to `DAGContext`, is it possible to call `initOutputInfo` and `initExecutorIdToJoinIdMap` in the constructor of `DAGContext`?